scene/gui/color_picker.cpp

```diff
@@ -1062,29 +1070,36 @@ void ColorPicker::_add_preset_pressed() {
 	emit_signal(SNAME("preset_added"), color);
 }
 
-void ColorPicker::_screen_pick_pressed() {
+void ColorPicker::_screen_pick_toggled(bool p_button_pressed) {
```

Comment:
This doesn't actually need to be converted, because `screen` will take `gui_input` until `screen` is hidden, that is, once sampling starts, there is no way to end sampling with a mouse button (You can use the `Esc` key, but not mouse clicks).

Reply:
Maybe this is a bug then, but it's fully possible to interact with the ColorPicker popup while screen is visible. It is currently not possible to pick a color from the popup, and clicking anywhere on the popup does not hide the screen. With this change it becomes possible to cancel the screen picker by pressing the same button again.

Reply:
Sorry, it seems that canceling sampling through the `Esc` key is not currently implemented. Might as well implement it now, it feels like it can be done once and for all.

Reply:
Absolutely, proper cancelling would be nice to have. I started looking into that too, but I didn't want to make much more changes while Vikita is working on her project. This was just to fix the issue that clicking the eyedropper icon again did not actually cancel picking.